src/plugins/ui_core_views/cell_evaluation/evaluator.ts

```diff
@@ -317,6 +317,7 @@ export class Evaluator {
     const formulaPosition = this.getters.getCellPosition(cellId);
 
     this.assertSheetHasEnoughSpaceToSpreadFormulaResult(formulaPosition, formulaReturn);
+    this.assertNoMergedCellsInSpreadZone(formulaPosition, formulaReturn);
```

Comment:
this check should be done after line 324 ( -> forEachSpreadPositionInMatrix(nbColumns, nbRows, this.updateSpreadRelation(formulaPosition)); )

Regardless of the presence of merges, we could want the formula spread positions to be updated.
@laa-odoo @anhe-odoo  Wouldn't it also be the case for formulas that spread out of the sheet?